src/connectors/infinite-hits/connectInfiniteHits.ts

```diff
@@ -262,30 +262,40 @@ const connectInfiniteHits: InfiniteHitsConnector = (
       },
 
       getWidgetState(uiState, { searchParameters }) {
-        const page = searchParameters.page || 0;
-
-        if (!hasShowPrevious || page === 0 || page + 1 === uiState.page) {
+        if (!hasShowPrevious) {
           return uiState;
         }
 
+        // The page in the UI state is incremented by one
+        // to expose the user value (not `0`).
+        const page = (searchParameters.page || 0) + 1;
+
         return {
           ...uiState,
-          page: page + 1,
+          page,
         };
       },
 
       getWidgetSearchParameters(searchParameters, { uiState }) {
-        if (!hasShowPrevious) {
-          return searchParameters;
+        let state = searchParameters;
+
+        if (escapeHTML) {
+          state = searchParameters.setQueryParameters(TAG_PLACEHOLDER);
         }
 
-        const uiPage = uiState.page;
+        if (hasShowPrevious) {
+          const uiPage = uiState.page;
 
-        if (uiPage) {
-          return searchParameters.setQueryParameter('page', uiPage - 1);
+          if (uiPage) {
+            // The page in the search parameters is decremented by one
+            // to get to the actual parameter value from the UI state.
+            state = searchParameters.setQueryParameter('page', uiPage - 1);
+          } else {
+            state = searchParameters.setQueryParameter('page', undefined);
```

Comment:
Is this branch even necessary? I left it here because I don't know why it was here before. When do we need to reset the `page` search parameter? (I suppose when option changes during the widget lifecycle but we don't support such a feature)

Reply:
I think this turns page 0 into undefined

Reply:
It's not necessary since we have to provide a default value, otherwise the widget is controlled by its parent (see [`pagination`](https://github.com/algolia/instantsearch.js/blob/f69ea2614a9d92fb48ba5535fae11537beb7818d/src/connectors/pagination/connectPagination.js#L182-L185)).